src/test/osd/osd-scrub-repair.sh

```diff
@@ -394,14 +485,1434 @@ function TEST_corrupt_scrub_replicated() {
+        },
+        {
+          "errors": [],
+          "osd": 1,
+          "selected": true,
+          "size": 7
+        }
+      ]
+    },
+    {
+      "all_shards": [
+        "stat_error"
+      ],
+      "errors": [],
+      "object": {
+        "locator": "",
+        "name": "OBJ13",
+        "nspace": "",
+        "snap": "head",
+        "version": 42
+      },
+      "shards": [
+        {
+          "errors": [
+            "stat_error"
+          ],
+          "osd": 0,
+          "selected": false,
+          "size": 18446744073709552000
+        },
+        {
+          "errors": [],
+          "osd": 1,
+          "selected": true,
+          "size": 7
+        }
+      ]
+    },
+    {
+    "all_shards": [
+      "oi_attr_missing",
+      "oi_attr_corrupted"
+    ],
+    "errors": [
+      "no_osd_to_compare"
+    ],
+    "object": {
+      "locator": "",
+      "name": "OBJ14",
+      "nspace": "",
+      "snap": "head",
+      "version": 0
+    },
+    "shards": [
+      {
+        "errors": [
+          "oi_attr_corrupted"
+        ],
+        "osd": 0,
+        "selected": false,
+        "size": 7
+      },
+      {
+        "errors": [
+          "oi_attr_missing"
+        ],
+        "osd": 1,
+        "selected": false,
+        "size": 7
+      }
+    ]
+  },
+  {
+    "all_shards": [
+      "oi_attr_missing"
+    ],
+    "errors": [],
+    "object": {
+      "locator": "",
+      "name": "OBJ15",
+      "nspace": "",
+      "snap": "head",
+      "version": 48
+    },
+    "shards": [
+      {
+        "errors": [],
+        "osd": 0,
+        "selected": true,
+        "size": 7
+      },
+      {
+        "errors": [
+          "oi_attr_missing"
+        ],
+        "osd": 1,
+        "selected": false,
+        "size": 7
+      }
+    ]
+  },
+  {
+      "all_shards": [
+        "missing"
+      ],
+      "errors": [],
+      "object": {
+        "locator": "",
+        "name": "OBJ3",
+        "nspace": "",
+        "snap": "head",
+        "version": 9
+      },
+      "shards": [
+        {
+          "errors": [],
+          "osd": 0,
+          "selected": true,
+          "size": 7
+        },
+        {
+          "errors": [
+            "missing"
+          ],
+          "osd": 1,
+          "selected": false
+        }
+      ]
+    },
+    {
+      "all_shards": [
+        "attr_value_mismatch",
+        "attr_name_mismatch"
+      ],
+      "errors": [],
+      "object": {
+        "locator": "",
+        "name": "OBJ8",
+        "nspace": "",
+        "snap": "head",
+        "version": 26
+      },
+      "shards": [
+        {
+          "attrs": [
+            {
+              "name": "_key1-OBJ8",
+              "value": "bad-val"
+            },
+            {
+              "name": "_key3-OBJ8",
+              "value": "val3-OBJ8"
+            }
+          ],
+          "errors": [
+            "attr_value_mismatch",
+            "attr_name_mismatch"
+          ],
```

Comment:
@athanatos This shard out of the 2 didn't have any errors before because it was the selected shard.  Now the selected shard gets marked with any errors that are discovered because we don't really know which is correct.